crates/c-api/include/wasmtime/component/instance.h

```diff
@@ -0,0 +1,27 @@
+#ifndef WASMTIME_COMPONENT_INSTANCE_H
+#define WASMTIME_COMPONENT_INSTANCE_H
+
+#ifdef WASMTIME_FEATURE_COMPONENT_MODEL
```

Comment:
I may have missed this on other headers, but I think for this to work reliably you'll want to `#include <wasmtime/conf.h>` in these headers too? (basically fix the case where this header, and just this header, are included)